### ISSUE TITLE
Custom loss function for the multiple responses/outputs

### ISSUE THREAD

**Issue by junjiwoo** [Microsoft/LightGBM#1985]
To be more simple, bivariate case, y_1 is 0 or 1, y_2 is continuous. Is it possible to customize the loss function to L = Gini + SSE? Or L = classification loss*V*regression loss where V can be a matrix for higher dimension.

**Comment by guolinke** [comment]
you can base on multi-class template, and write the customed objective functions.

**Comment by guolinke** [comment]
Hi @junjiwoo 
I think you can use python package to achieve this, by the following processing.
1.  construct two lgb.Dataset, the X is the same, but with different label.
2. construct two lgb.Booster, each accepts one lgb.Dataset
3. write two objective functions, each for a lgb.Booster
4. use lgb.Booster.update https://lightgbm.readthedocs.io/en/latest/Python-API.html#lightgbm.Booster.update , to train the booster. You can call this in the loop for mulitply iterations.

BTW, for the step 4, you may need to some shares between two datasets, as your objective function needs the information from both two data.

**Comment by junjiwoo** [comment]
Thank you for your prompt response. I will follow your advice. I saw many related questions and requests. To the best of my knowledge, nobody implemented this yet. (gbm, mvtboost, xgboost, h2o)

**Comment by StrikerRUS** [comment]
We already have an issue related to multi-output: #524. I think it's better to keep the discussion in one place.
